ibm/resource_ibm_cis_edge_functions_action.go

```diff
@@ -0,0 +1,187 @@
+package ibm
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/IBM/go-sdk-core/v3/core"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	cisEdgeFunctionsActionScriptName = "script_name"
+	cisEdgeFunctionsActionScript     = "script"
+	cisEdgeFunctionsActionScriptFile = "/tmp/script.js"
+)
+
+func resourceIBMCISEdgeFunctionsAction() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMCISEdgeFunctionsActionCreate,
+		Read:     resourceIBMCISEdgeFunctionsActionRead,
+		Update:   resourceIBMCISEdgeFunctionsActionUpdate,
+		Delete:   resourceIBMCISEdgeFunctionsActionDelete,
+		Exists:   resourceIBMCISEdgeFunctionsActionExists,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "CIS Intance CRN",
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "CIS Domain ID",
+				DiffSuppressFunc: suppressDataDiff,
+			},
+			cisEdgeFunctionsActionScriptName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Edge function action script name",
+			},
+			cisEdgeFunctionsActionScript: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Edge function action script",
+			},
+		},
+	}
+}
+
+func resourceIBMCISEdgeFunctionsActionCreate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisEdgeFunctionClientSession()
+	if err != nil {
+		return err
+	}
+
+	crn := d.Get(cisID).(string)
+	zoneID, _, err := convertTftoCisTwoVar(d.Get(cisDomainID).(string))
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+
+	scriptName := d.Get(cisEdgeFunctionsActionScriptName).(string)
+	script := d.Get(cisEdgeFunctionsActionScript).(string)
+	err = createScriptFile(script)
+	if err != nil {
+		return fmt.Errorf("Error: not able to create the action script file")
+	}
+	reader, err := os.Open(cisEdgeFunctionsActionScriptFile)
+	if err != nil {
+		return fmt.Errorf("Error: not able to read the action script file")
+	}
+
+	opt := cisClient.NewUpdateEdgeFunctionsActionOptions(scriptName)
+	opt.SetEdgeFunctionsAction(reader)
+
+	_, _, err = cisClient.UpdateEdgeFunctionsAction(opt)
+	if err != nil {
+		return fmt.Errorf("Error: %v", err)
+	}
+	d.SetId(convertCisToTfThreeVar(scriptName, zoneID, crn))
+	return resourceIBMCISEdgeFunctionsActionRead(d, meta)
+}
+
+func resourceIBMCISEdgeFunctionsActionUpdate(d *schema.ResourceData, meta interface{}) error {
+	// No need to have upate function, since edge function script is going to be uploaded as a file
+	// so calling create method here.
```

Comment:
please check for hasChange of `cisEdgeFunctionsActionScript ` and return create inside the condition block

Reply:
done